staging/src/k8s.io/kube-gen/cmd/conversion-gen/generators/conversion.go

```diff
@@ -338,36 +338,42 @@ func (e equalMemoryTypes) Equal(a, b *types.Type) bool {
 	if equal, ok := e[conversionPair{b, a}]; ok {
 		return equal
 	}
-	result := e.equal(a, b)
+	result := e.equal(a, b, alreadyVisitedTypes)
```

Comment:
We can't do that because we need the populated map in line 366, 372, 374 and 376 below. If we create an empty map here and those lines are executed, they will see the type as "not visited" - which will defeat our purpose. :)

Also, tried it out. Gives the `goroutine stack exceeds 1000000000-byte limit` error.